react-native/ios/RealmReact/RealmReact.mm

```diff
@@ -55,20 +56,21 @@ - (JSGlobalContextRef)ctx;
     id rctJSContext = object_getIvar(executor, contextIvar);
     if (!rctJSContext && create) {
         Class RCTJavaScriptContext = NSClassFromString(@"RCTJavaScriptContext");
-        NSMethodSignature *signature = [RCTJavaScriptContext instanceMethodSignatureForSelector:@selector(initWithJSContext:)];
-        assert(signature);
 
-        // React Native 0.14+ expects a JSContext here, but we also support 0.13.x, which takes a JSGlobalContextRef.
-        if (!strcmp([signature getArgumentTypeAtIndex:2], "@")) {
-            JSContext *context = [[JSContext alloc] init];
-            rctJSContext = [[RCTJavaScriptContext alloc] initWithJSContext:(void *)context];
+        NSMethodSignature *signature = [RCTJavaScriptContext instanceMethodSignatureForSelector:@selector(initWithJSContext:onThread:)];
+        if (signature) {
+            // for RN 0.28.0+
+            rctJSContext = [executor context];
```

Comment:
I agree that these changes can be cleaned up a bit - will incorporate some of your suggestions into the pr. Based on @jerson 's testing though we will still need to populate the `_jscWrapper` ivar as well as this is expected to exist on calls to `-context`. Alternatively we can revert to the old methodology we were using where we would create and set the context manually rather than depending on `-context`. Will hopefully get the pr updated today so you guys can test this out as I am still unable to trigger this code path.

In the longer term we need to be making a pr against ReactNative to remove all dependencies on RN implementation details. We tried this at one point the but unfortunately there was a race condition with the changes that were merged into RN, so we had to revert to this strategy to initialize the Realm apis.